minio/operator
# Single node stateful-kes integration

Steps to test :-

STEP 1: Install operator with the dev image
```sh
./kubectl-minio init --image minio/operator:v4.4.21-60-gb0ad327 --default-kes-image minio/kes:v0.20.0
```

STEP 2: Generate a unseal key for kes

```sh
cat /dev/urandom | head -c 32 | base64
```

and set it in `examples/kustomization/tenant-stateful-kes-encryption/tenant.yaml`

STEP 3: Configure volumeClaimTemplate section in "stateful-kes". Use directpv storage class if directpv is configured

STEP 4: Create the tenant

```sh
kustomize build examples/kustomization/tenant-stateful-kes-encryption | kubectl apply -f -
```

STEP 5: Once the pods are up, get the enclave-admin.key and enclave-admin.crt from the secrets

```sh
kubectl get secret storage-stateful-kes-enclave-admin-config -n minio-tenant -o yaml
```
(NOTE: base64 decode the private.key and public.crt and place it in respective files enclave-admin.key and enclave-admin.crt)

STEP 6: Check if the key is created

```sh
export KES_SERVER=<KES_POD_ENDPOINT>
export KES_CLIENT_KEY=enclave-admin.key
export KES_CLIENT_CERT=enclave-admin.crt

kes key ls -k
```
